### PR TITLE
Refactor the code and fix #9.

### LINKED ISSUE #9: utf support
hi

i have this result when running:

```

~/workspace/MyAppNap> /usr/bin/python myAppNap.py Zdjęcia
Invalid app name, will not suspend/resume anything
        Will monitor apps in focus, switch to your desired app to see valid name
Currently focused on Terminal
myAppNap.py:53: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
  if last_active_app == desiredApp:

Currently focused on Zdjęcia # its photos app in polish

Currently focused on Terminal


^C
Exiting script
~/workspace/MyAppNap> /usr/bin/python myAppNap.py Trello 
Monitoring Trello, with PIDs: ['2018', '2019', '2028', '2030'] # multiple PIDs, not working then
Currently focused on Terminal
Stopping Trello
Traceback (most recent call last):
  File "myAppNap.py", line 64, in <module>
    os.kill(pid, signal.SIGSTOP)
TypeError: an integer is required
```

### PR DESCRIPTION
**Pull request by PurpleMyst**
I've moved everything from the top-level into a `main` function, made the script version-agnostic and used double quotes consistently.

I'd like to hear your feedback. :)

### CODE REVIEW

File: myAppNap.py [hunk 1]
```diff
@@ -1,72 +1,96 @@
...
-''' This script automatically finds PIDs for desiredApp
-    and suspend those processes whenever desired app  is not in focus. 
-    This script is meant to stay running and polls window focus every 1 second
-    On receiving keyboard interrupt, will resume
-'''
-from datetime import datetime
-from time import sleep
-import sys
-import subprocess
+#!/usr/bin/env python
```
Comment (joesanford) [inline myAppNap.py]: `/usr/bin/python` should be preferred as this script requires using pre-installed Python.

File: myAppNap.py [hunk 2]
```diff
@@ -1,72 +1,96 @@
...
-                    for pid in pids:
-                        os.kill(pid, signal.SIGSTOP)
-        sleep(1)
-except KeyboardInterrupt:
-    print '\nExiting script'
+        active_app_name = active_app["NSApplicationName"]
+
+        # We only want to check if something is an `unicode` if we're in
+        # Python 2. Python 3 doesn't have this distinction anymore.
+        if sys.version_info.major < 3 and isinstance(active_app_name, unicode):
```
Comment (joesanford) [inline myAppNap.py]: This changes functionality and goes beyond a refactor. Also, this script is only run by pre-installed Python on macOS which is Python 2

File: myAppNap.py (hunk 1 above)
Comment (PurpleMyst) [reply myAppNap.py]: If I recall correctly, you can install the system packages on a non-pre-installed python, can you not?

File: myAppNap.py (hunk 2 above)
Comment (PurpleMyst) [reply myAppNap.py]: This PR evolved from a refactor into fixing a few issues, such as #9. I should change the title.

Comment (joesanford) [reply myAppNap.py]: Generally you should split PR's then.

Comment (PurpleMyst) [reply myAppNap.py]: That is true, however wouldn't a split PR act on the non-refactored code-base?

File: myAppNap.py (hunk 1 above)
Comment (joesanford) [reply myAppNap.py]: This requires:
- A proper `requirements.txt`
- A change in the logic for importing AppKit
- Testing with `PyObjC`

File: myAppNap.py [hunk 3]
```diff
@@ -1,72 +1,96 @@
...
+    sys.exit(1)
+
 
-def get_pid(name):
+def get_pids(name):
     try:
-        result = subprocess.check_output(['pgrep '+name], shell=True)
-        return result.strip().split('\n')
+        # XXX: I've removed `shell=True` here. You should verify that
+        # it still works, seeing as I sadly can't test anything on OSX.
```
Comment (joesanford) [inline myAppNap.py]: Why is this part of a refactor? More functional changes.

Comment (PurpleMyst) [reply myAppNap.py]: I believe that adding/removing `shell=True` in that location should not change anything.

Comment (joesanford) [reply myAppNap.py]: See previous comments re: refactor vs. functional changes.

**Pull request closed without merging**